packages/protocol/contracts/adminnable/SelfAdminnable.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.6;
+
+import "./interfaces/ISelfAdminnable.sol";
+
+/// @title Contract that implements multiple levels of admins for multiple
+/// "adminned" addresses independently
+contract SelfAdminnable is ISelfAdminnable {
```

Comment:
Same here,  should this be an abstract contracts with a non implemented getRank() function?